src/main/java/bio/terra/cli/Workspace.java

```diff
@@ -0,0 +1,212 @@
+import bio.terra.cli.serialization.disk.DiskWorkspace;
+import bio.terra.cli.service.WorkspaceManagerService;
+import bio.terra.workspace.model.WorkspaceDescription;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Internal representation of a workspace. An instance of this class is part of the current context
+ * or state.
+ */
+public class Workspace {
+  private static final Logger logger = LoggerFactory.getLogger(Workspace.class);
+
+  private UUID id;
+  private String name; // not unique
+  private String description;
+  private String googleProjectId;
+
+  // name of the server where this workspace exists
+  private String serverName;
+
+  // email of the user that loaded the workspace to this machine
+  private String userEmail;
+
+  // list of resources (controlled & referenced)
+  private List<Resource> resources;
+
+  /** Build an instance of this class from the WSM client library WorkspaceDescription object. */
+  private Workspace(WorkspaceDescription wsmObject) {
+    this.id = wsmObject.getId();
+    this.name = wsmObject.getDisplayName() == null ? "" : wsmObject.getDisplayName();
+    this.description = wsmObject.getDescription() == null ? "" : wsmObject.getDescription();
+    this.googleProjectId =
+        wsmObject.getGcpContext() == null ? null : wsmObject.getGcpContext().getProjectId();
+    this.serverName = Context.getServer().getName();
+    this.userEmail = Context.requireUser().getEmail();
+    this.resources = new ArrayList<>();
+  }
+
+  /** Build an instance of this class from the serialized format on disk. */
+  public Workspace(DiskWorkspace configFromDisk) {
+    this.id = configFromDisk.id;
+    this.name = configFromDisk.name;
+    this.description = configFromDisk.description;
+    this.googleProjectId = configFromDisk.googleProjectId;
+    this.serverName = configFromDisk.serverName;
+    this.userEmail = configFromDisk.userEmail;
+    this.resources =
+        configFromDisk.resources.stream()
+            .map(DiskResource::deserializeToInternal)
+            .collect(Collectors.toList());
+  }
+
+  /**
+   * Create a new workspace and set it as the current workspace.
+   *
+   * @param name optional display name
+   * @param description optional description
+   */
+  public static Workspace create(String name, String description) {
+    // call WSM to create the workspace object and backing Google context
+    WorkspaceDescription createdWorkspace =
+        new WorkspaceManagerService().createWorkspace(name, description);
+    logger.info("Created workspace: {}", createdWorkspace);
+
+    // convert the WSM object to a CLI object
+    Workspace workspace = new Workspace(createdWorkspace);
+
+    // fetch the pet SA credentials for the user + this workspace
+    Context.requireUser().fetchPetSaCredentials();
```

Comment:
I do it here so we have the pet SA key file stored locally before the user tries to run an app in the workspace. This is basically paying the cost of a SAM round-trip ahead of time, instead of slowing down an app call.

If the SAM connection wonks out, it will throw an error. I'll move it to the end of the method call so we still update the context regardless.